pallets/system/src/header_builder.rs

```diff
@@ -263,3 +247,178 @@ impl Decode for AvailExtrinsic {
 		})
 	}
 }
+
+fn build_data_root(app_ext: Vec<AppExtrinsic>) -> [u8; 32] {
+	let ext: Vec<Vec<u8>> = app_ext.iter().map(|x| x.data.clone()).collect();
+	let avail_ext = ext
+		.iter()
+		.filter_map(|e| AvailExtrinsic::decode(&mut &e[..]).ok())
+		.collect::<Vec<_>>();
+	let data_root: [u8; 32] = if avail_ext.len() > 0 {
+		let leaves: Vec<[u8; 32]> = avail_ext.iter().map(|e| Sha256::hash(&e.data)).collect();
+		let tree = MerkleTree::<Sha256>::from_leaves(&leaves);
+		tree.root().expect("Data Root computation failed")
+	} else {
+		Default::default()
+	};
+	data_root
+}
```

Comment:
@MiguelDD1 We want the data root to be the Merkle root of only `SubmitData()` calls, i.e., the leaves of the tree should contain only the data posted by users and *not* 
- all the extrinsics in the block
- encoded extrinsics

This is because, on Ethereum (or destination chain), we want to give proofs of data inclusion. If we push `extrinsic_root()`, then
- Verifying proof of inclusion is costly (Blake2 is gas expensive)
- Even if we create SHA/keccak based merkle tree, still to show inclusion we need to
  - give the entire encoded extrinsic -> this defeats purpose of Validiums
  - verify proof of inclusion
  - decode the extrinsic -> gas expensive

Reply:
@kroos47 Let's change the root field to `H256` and update/test the LC, explorer and test scripts. Should work without much trouble IMO.

Reply:
done!